### DISCUSSION TITLE
Error: Input file has corrupt header: heif: Invalid input: No 'hvcC' box (2.106)

### DISCUSSION THREAD

**Discussion by V0lantis** [immich-app/immich#14490] [Q&A]
Hello team,

I'm using immich with [Unraid docker-compose](https://immich.app/docs/install/unraid/) and I've noticed some error logs in my `immich_server` container:

```log
[Nest] 7  - 12/05/2024, 12:00:00 AM   ERROR [Microservices:JobService] Unable to run job handler (thumbnailGeneration/generate-thumbnails): Error: Input file has corrupt header: heif: Invalid input: No 'hvcC' box (2.106)
[Nest] 7  - 12/05/2024, 12:00:00 AM   ERROR [Microservices:JobService] Error: Input file has corrupt header: heif: Invalid input: No 'hvcC' box (2.106)
    at Sharp.toBuffer (/usr/src/app/node_modules/sharp/lib/output.js:163:17)
    at MediaRepository.decodeImage (/usr/src/app/dist/repositories/media.repository.js:57:68)
    at MediaService.generateImageThumbnails (/usr/src/app/dist/services/media.service.js:158:63)
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async MediaService.handleGenerateThumbnails (/usr/src/app/dist/services/media.service.js:108:25)
    at async JobService.onJobStart (/usr/src/app/dist/services/job.service.js:151:28)
    at async EventRepository.onEvent (/usr/src/app/dist/repositories/event.repository.js:122:13)
    at async Worker.processJob (/usr/src/app/node_modules/bullmq/dist/cjs/classes/worker.js:394:28)
    at async Worker.retryIfFailed (/usr/src/app/node_modules/bullmq/dist/cjs/classes/worker.js:581:24)
[Nest] 7  - 12/05/2024, 12:00:00 AM   ERROR [Microservices:JobService] Object:
{
  "id": "ecd9121f-c9c5-4cff-a077-bc96976df08b"
}
```

So far, I haven't found any issues which seem related to it on [immich issues](https://github.com/immich-app/immich/issues). Have I overlooked something? How can I solve this error message?

Thank you so much for the hard work !

**Comment by anyshpm** [comment]
I'm encountering the same error. It seems the original file is corrupted. How do I repair it?

**Reply by krypton9208** [reply]
I just EDIT photo in Ios app (build in photo), my photos was "live" soo i just change te keyframe and after that i upload to immich photo once again

**Reply by HikoSwallow** [reply]
我也遇到了同样的问题。图片是我从网络上下载的，我不确定它们是否来自于IOS系统，但是通过上述回复我想问题出在图片的编码方式上，于是我想通过转换图片格式的方法去消除产生错误的元数据。
我使用python代码尝试打开图片，但是都失败了。最后我使用了格式工厂对图片格式进行了转化，解决了这一报错。

**Comment by NightMean** [comment]
I actually have the same issue but this is an iOS issue where for some unknown reason, iOS very rarely produces a broken HEIF file but I have experienced this with multiple versions. 
Unfortunately immich (v.135.3) currently does not show this as Failed transcode within the UI and I had to dig into the logs to see this issue. 
Hopefully this can be fixed later on as there is ability to show Failed jobs already within the UI (even though not the exact job that failed)

**Comment by harvasyuk** [comment]
Here's an example of an image that breaks immich_server with an error "Input file has corrupt header immich".
https://unsplash.com/photos/lighted-house-in-city-near-glacier-mountain-at-nighttime-_7hiYkKVmsk

To reproduce the issue, just download an image from Unsplash and try uploading it using a browser. In my case, the server immediately crashes. Similar situation with uploading some edited images from iOS (edited screenshots or photos in lightroom) - the server cannot create previews for them.
